nconf/tls.go

```diff
@@ -21,88 +24,89 @@ type TLSConfig struct {
 }
 
 func (cfg TLSConfig) TLSConfig() (*tls.Config, error) {
-	var tlsconf *tls.Config
 	var err error
-	if cfg.Cert != "" && cfg.Key != "" {
-		tlsconf, err = LoadFromValues(cfg.Cert, cfg.Key, cfg.CA)
-	} else if cfg.CertFile != "" && cfg.KeyFile != "" {
-		tlsconf, err = LoadFromFiles(cfg.CertFile, cfg.KeyFile, cfg.CAFiles)
-	}
 
-	if err != nil {
-		return nil, err
+	tlsConf := &tls.Config{
+		MinVersion:         tls.VersionTLS12,
+		InsecureSkipVerify: cfg.Insecure,
 	}
 
-	if tlsconf != nil {
-		tlsconf.InsecureSkipVerify = cfg.Insecure
+	// Load CA
+	if cfg.CA != "" {
+		logrus.Debugf("Loading CA value")
+		tlsConf.RootCAs, err = LoadCAFromValue(cfg.CA)
+	} else if len(cfg.CAFiles) > 0 {
+		logrus.WithField("ca_files", cfg.CAFiles).Debugf("Loading CA files")
+		tlsConf.RootCAs, err = LoadCAFromFiles(cfg.CAFiles)
+	} else {
+		logrus.Debugf("No CA provided. Use system certs")
+		tlsConf.RootCAs, err = x509.SystemCertPool()
 	}
 
-	return tlsconf, nil
-}
+	if err != nil {
+		return nil, errors.Wrap(err, "Error setting up Root CA pool")
+	}
 
-func LoadFromValues(certPEM, keyPEM, ca string) (*tls.Config, error) {
-	var pool *x509.CertPool
-	// If no CA cert if provided, use system pool
-	if ca == "" {
-		p, err := x509.SystemCertPool()
-		if err != nil {
-			return nil, err
-		}
-		pool = p
-	} else {
-		pool = x509.NewCertPool()
-		if !pool.AppendCertsFromPEM([]byte(ca)) {
-			return nil, fmt.Errorf("Failed to add CA cert")
-		}
+	// Load Certs if any
+	var cert tls.Certificate
+	err = nil
```

Comment:
This shouldn't be needed it - you return if it was an error before

Reply:
argh, of course. Forgot to delete this